kube/client/client.go

```diff
@@ -0,0 +1,92 @@
+// +build kube
+
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/docker/compose-cli/api/compose"
+	"github.com/docker/compose-cli/utils"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+)
+
+// KubeClient holds Kubernetes config and client
+type KubeClient struct {
+	config genericclioptions.RESTClientGetter
+	client *kubernetes.Clientset
+}
+
+// NewKubeClient intializes a Kubernetes client
+func NewKubeClient(config genericclioptions.RESTClientGetter) (*KubeClient, error) {
+	restConfig, err := config.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &KubeClient{
+		config: config,
+		client: clientset,
+	}, nil
+}
+
+// GetLogs retrieves pod logs
+func (c *KubeClient) GetLogs(ctx context.Context, projectName string, consumer compose.LogConsumer, follow bool) error {
+	namespace, _, err := c.config.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	pods, err := c.client.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", compose.ProjectTag, projectName),
+	})
+	if err != nil {
+		return err
+	}
+
+	eg, ctx := errgroup.WithContext(ctx)
+
+	for _, pod := range pods.Items {
+
+		request := c.client.CoreV1().Pods(namespace).GetLogs(pod.Name, &corev1.PodLogOptions{Follow: follow})
```

Comment:
Maybe we should avoid streaming logs for all services if we are filtering on services at the consumer level (just optimizing, but this looks easy enough)

Reply:
Can be looked at in a separate PR, to merge this one quicker and avoid conflicts on baseic KubeClient struct